projects/plugins/jetpack/modules/search/class-jetpack-search-customberg.php

```diff
@@ -74,13 +79,33 @@ public function jetpack_search_admin_page() {
 		// TODO: Spin this function off into a static helper function in a helper class for code reuse.
 		$static_url = apply_filters( 'jetpack_static_url', '//en.wordpress.com/i/loading/loading-64.gif' );
 		?>
-			<div id="jp-search-customization" class="jp-search-customization-dashboard">
-				<div class="hide-if-no-js"><img class="jp-search-loader" width="32" height="32" alt="<?php esc_attr_e( 'Loading&hellip;', 'jetpack' ); ?>" src="<?php echo esc_url( $static_url ); ?>" /></div>
-				<div class="hide-if-js"><?php esc_html_e( 'Your Search customization page requires JavaScript to function properly.', 'jetpack' ); ?><div />
+			<div id="jp-search-customization" class="jp-search-customization-dashboard" style="height: calc(100vh - 100px);">
+				<div class="hide-if-no-js" style="height: 100%;">
+					<img class="jp-search-loader" width="32" height="32" alt="<?php esc_attr_e( 'Loading&hellip;', 'jetpack' ); ?>" src="<?php echo esc_url( $static_url ); ?>" style="
+						position: absolute;
+						left: 50%;
+						top: 50%;
+					"/>
+				</div>
+				<div class="hide-if-js"><?php esc_html_e( 'Your Search customization page requires JavaScript to function properly.', 'jetpack' ); ?></div>
 			</div>
 		<?php
 	}
 
+	/**
+	 * Redirects to the Customizer if Customberg is not supported by the current host.
+	 */
+	public function add_redirect_if_necessary() {
+		// Add a JS redirect if Customberg is not supported.
+		if ( ! $this->wp_supports_customberg() ) {
+			?>
+				<script>
+					window.location.href="/wp-admin/customize.php?autofocus[section]=jetpack_search";
```

Comment:
probably need to use `admin_url()` somehow for the sake of WP installed in a sub directory?

Reply:
Fixed. Debugging this made me realize that the Customberg redirect hook was being used incorrectly, so that has been fixed as well.